content/rest-api/index.textile

```diff
@@ -660,57 +815,103 @@ bc[json]. [{
   channel: <string>
 }]
 
-h3(#delete-device-registration). Unregister device for push notifications
-
-h6. DELETE rest.ably.io/push/deviceRegistrations
+h3(#get-channels). List all channels with at least one subscribed device
 
-Unregister device. All its subscriptions will be removed.
+h6. GET rest.ably.io/push/channels
 
 Example request:
 
-bc[sh]. curl -X DELETE https://rest.ably.io/push/deviceRegistrations?deviceId=01ARZ3NDEKTSV4RRFFQ69G5FAV \
+bc[sh]. curl https://rest.ably.io/push/channelSubscriptions \
  -u "{{API_KEY}}"
 
 h5. Parameters
 
-- deviceId := Filter to restrict to subscriptions for that @devideId@. Cannot be set when @clientId@ is set.
-- clientId := Filter to restrict to subscriptions associated with that @clientId@. Cannot be set when @deviceId@ is set.
+None
 
 h5. Options
 
 - Content-Type := not applicable
-- Auth required := yes ("basic":#basic-authentication or "token":#token-authentication)
+- Accept := @application/json@ by default, or @application/x-msgpack@, @text/html@
+- Auth required := yes ("basic":#basic-authentication or "token":#token-authentication with @push-admin@ capability)
 
 h5. Returns
 
-A successful request returns nothing.
+A successful request returns a "paginated response":#pagination of:
 
-An unsuccessful request returns an error.
+bc[json]. [<string, channel name>]
 
-h3(#delete-channel-subscription). Unsubscribe from push notifications
+h3(#push-publish). Publish a push notification to a single device
 
-h6. DELETE rest.ably.io/push/channelSubscriptions
+Convenience endpoint to deliver a push notification payload to a single device.
 
-Stop receiving push notifications when push messages arrive to channels.
+Note that this bypasses. If you want to send a push notification to multiple devices, or use a more flexible publish-subscribe architecture so that you don't need to know about recipient devices's details, you should look at "registering devices for push":#post-device-registration, then "subscribe them to channels":#post-channel-subscription, and then "send messages to the channels with push payloads":#message-extras-push.
+
+h6. POST rest.ably.io/push/publish
+
+The request body is an object of the form:
+
+bc[json]. {
+  transportType: <apns, fcm, gcm, web>
+  recipient: <object, transport-specific>
+  <... rest of the fields just like a normal push-enabled message ...>
+}
+
+The @recipient@ field depends on the underlying transport used to deliver the message.
+
+For APNs:
+
+bc[json]. {
+  deviceToken: <string>
+}
+
+For GCM/FCM:
+
+bc[json]. {
+  registrationToken: <string>
+}
+
+For web:
+
+bc[json]. {
+  targetUrl: <string>
+  encryptionKey: <string>
+}
```

Comment:
Ah, I thought direct push was just for specific devices, completely bypassing registrations. OK, I'll add that.